### PR TITLE
decomp: finish `mother-spider`

### PR DESCRIPTION
**Pull request by xTVaser**

### CODE REVIEW

File: goal_src/levels/maincave/mother-spider.gc [hunk 1]
```diff
@@ -5,3 +5,2134 @@
...
+  :trans
+  (behavior ()
+    (local-vars (at-0 int))
+    (rlet ((vf0 :class vf)
+           (vf1 :class vf)
+           (vf2 :class vf)
+           )
+      (init-vf0-vector)
+      (+! (-> self transv y) (* (-> self gravity) (-> *display* seconds-per-frame)))
+      (let ((gp-0 (new 'stack-no-clear 'collide-mesh-cache-tri))
```
Comment (water111) [inline goal_src/levels/maincave/mother-spider.gc:105]: probably should be `collide-tri-result`

File: goal_src/levels/maincave/mother-spider.gc [hunk 2]
```diff
@@ -5,3 +5,2134 @@
...
+
+(defmethod STUB-28 mother-spider ((obj mother-spider))
+  0
+  (none)
+  )
+
+(defmethod TODO-RENAME-24 mother-spider ((obj mother-spider))
+  (cond
+    ((and (-> obj draw shadow) (zero? (-> obj draw cur-lod)) (logtest? (-> obj draw status) 8))
+     (let ((s5-0 (new 'stack-no-clear 'collide-mesh-cache-tri))
```
Comment (water111) [inline goal_src/levels/maincave/mother-spider.gc:448]: should probably be `collide-tri-result`

File: goal_src/levels/maincave/mother-spider.gc [hunk 3]
```diff
@@ -5,3 +5,2134 @@
...
+            (>= (- (-> *display* base-frame-counter) (the-as int (-> self last-spit-time))) 900)
+            (>= (- (-> *display* base-frame-counter) (the-as int (-> self last-player-in-air-time))) 18)
+            (>= (-> self max-spit-xz-dist) (vector-vector-xz-distance (-> self root-override trans) (target-pos 0)))
+            )
+      (let ((gp-2 (new 'stack-no-clear 'vector))
+            (s5-2 (new 'stack-no-clear 'vector))
+            )
+        (set! (-> gp-2 quad) (-> self root-override trans quad))
+        (set! (-> gp-2 w) 4096.0)
+        (when (sphere-in-view-frustum? (the-as sphere gp-2))
```
Comment (water111) [inline goal_src/levels/maincave/mother-spider.gc:1155]: interesting, I think this means the spider won't shoot if you can't see it.

File: goal_src/levels/maincave/mother-spider.gc [hunk 4]
```diff
@@ -5,3 +5,2134 @@
...
+          )
+        (vector-float*! s2-1 s1-0 f30-0)
+        (dummy-35 (-> obj nav) (-> obj nav travel) (-> obj root-override trans) s2-1 s1-0 f30-0)
+        )
+      (dummy-24 (-> obj nav) f30-0 (the-as clip-travel-vector-to-mesh-return-info #f))
+      )
+    (cond
+      ((>= (vector-xz-length (-> obj nav travel)) 4096.0)
+       (vector+! arg0 (-> obj root-override trans) (-> obj nav travel))
+       (let ((s2-2 (new 'stack-no-clear 'collide-mesh-cache-tri)))
```
Comment (water111) [inline goal_src/levels/maincave/mother-spider.gc:1552]: `collide-tri-result`

File: goal_src/levels/maincave/mother-spider.gc [hunk 5]
```diff
@@ -5,3 +5,2134 @@
...
+             )
+           )
+         )
+       )
+      (else
+        0
+        )
+      )
+    )
+  (let ((t9-11 (method-of-object (-> obj nav) dummy-17)))
```
Comment (water111) [inline goal_src/levels/maincave/mother-spider.gc:1581]: missing args

File: goal_src/levels/maincave/mother-spider.gc [hunk 6]
```diff
@@ -5,3 +5,2134 @@
...
+    (-> obj root-override trans)
+    (t9-11)
+    )
+  (let ((a1-12 (new 'stack-no-clear 'vector))
+        (s3-2 (new 'stack-no-clear 'collide-mesh-cache-tri))
```
Comment (water111) [inline goal_src/levels/maincave/mother-spider.gc:1586]: `collide-tri-result`

**Review by water111 — COMMENTED** [review]
just a few stack types

File: goal_src/levels/maincave/mother-spider.gc (hunk 3 above)
Comment (xTVaser) [reply goal_src/levels/maincave/mother-spider.gc:1155]: that wouldn't be too surprising -- that's a common thing across all the jak games.  It'll be interesting to see if this same pattern is used in the future

Comment (ManDude) [reply goal_src/levels/maincave/mother-spider.gc:1155]: I'm pretty sure looking away from a spider is a strat used in speedruns when shooting the gnawers.

**Pull request merged**